services/slam/builtin/orbslam_int_test.go

```diff
@@ -56,43 +103,162 @@ func TestOrbslamIntegrationExample(t *testing.T) {
 	for {
 		line, err := logReader.ReadString('\n')
 		test.That(t, err, test.ShouldBeNil)
-		if strings.Contains(line, "Passed image to SLAM") {
+		if strings.Contains(line, "Finished processing offline images") {
 			break
 		}
+		test.That(t, strings.Contains(line, "Fail to track local map!"), test.ShouldBeFalse)
 	}
 
-	// Test position and map
-	position, err := svc.Position(context.Background(), "test")
-	test.That(t, err, test.ShouldBeNil)
-	t.Logf("Position point: (%v, %v, %v)",
-		position.Pose().Point().X, position.Pose().Point().Y, position.Pose().Point().Z)
-	t.Logf("Position orientation: RX: %v, RY: %v, RY: %v, Theta: %v",
-		position.Pose().Orientation().AxisAngles().RX,
-		position.Pose().Orientation().AxisAngles().RY,
-		position.Pose().Orientation().AxisAngles().RZ,
-		position.Pose().Orientation().AxisAngles().Theta)
-	actualMIME, _, pointcloud, err := svc.GetMap(context.Background(), "test", "pointcloud/pcd", nil, false)
+	testPositionAndMap(t, svc)
+
+	// Wait for the final map to be saved
+	for {
+		line, err := logReader.ReadString('\n')
+		test.That(t, err, test.ShouldBeNil)
+		if strings.Contains(line, "Finished saving final map") {
+			break
+		}
+	}
+
+	// Close out slam service
+	test.That(t, utils.TryClose(context.Background(), svc), test.ShouldBeNil)
+	// Don't clear out the directory, since we will re-use the maps for the next run
+	closeOutSLAMService(t, "")
+
+	// Remove existing images, but leave maps and config (so we keep the vocabulary file).
+	// Orbslam will use the most recent config.
+	test.That(t, os.RemoveAll(name+"/data"), test.ShouldBeNil)
+
+	// Test online mode using the map generated in the offline test
+	t.Log("Testing online mode with saved map")
+
+	attrCfg = &builtin.AttrConfig{
+		Algorithm: "orbslamv3",
+		Sensors:   []string{"orbslam_int_color_camera", "orbslam_int_depth_camera"},
+		ConfigParams: map[string]string{
+			"mode":              "rgbd",
+			"orb_n_features":    "1000",
+			"orb_scale_factor":  "1.2",
+			"orb_n_levels":      "8",
+			"orb_n_ini_th_fast": "20",
+			"orb_n_min_th_fast": "7",
+			"debug":             "true",
+		},
+		DataDirectory: name,
+		MapRateSec:    -1,
+	}
+
+	// Release a pair of camera images for service validation
+	releaseImages()
+	// Create slam service using a real orbslam binary
+	svc, err = createSLAMService(t, attrCfg, golog.NewTestLogger(t), true, true)
 	test.That(t, err, test.ShouldBeNil)
-	test.That(t, actualMIME, test.ShouldResemble, "pointcloud/pcd")
-	t.Logf("Pointcloud points: %v", pointcloud.Size())
+
+	// Make sure we initialize from a saved map
+	logReader = svc.(internal.Service).GetSLAMProcessBufferedLogReader()
+	for {
+		line, err := logReader.ReadString('\n')
+		test.That(t, err, test.ShouldBeNil)
+		if strings.Contains(line, "Initialization of Atlas from file") {
+			break
+		}
+		test.That(t, strings.Contains(line, "Initialization of Atlas from scratch"), test.ShouldBeFalse)
```

Comment:
It's possible for this to fail due to https://viam.atlassian.net/browse/DATA-467. If we see that happen in CI, we may need to fix the bug sooner.